model/lifecycle.go

```diff
@@ -389,6 +389,23 @@ func RestartVersion(versionId string, taskIds []string, abortInProgress bool, ca
 	for _, t := range finishedTasks {
 		if !t.IsPartOfSingleHostTaskGroup() { // for single host task groups we don't archive until fully restarting
 			if err = t.Archive(); err != nil {
+				type idAndExecution struct {
+					Id        string `json:"id"`
+					Execution int    `json:"execution"`
+				}
+				dbTasks := []idAndExecution{}
+				for _, dbTask := range finishedTasks {
```

Comment:
archive destructively modifies execution; would it be bad for performance to get this information before ever going into the loop? if so, then maybe just include results for up to the failing task, or else the data in this list is inconsistent and that might be confusing

Reply:
I changed the field name so it's a bit clearer but I think it's fine

Reply:
Well I was talking about dbTasks: if there are 5 tasks,  and we've successfully archived task one and two, then task 3 fails, task 1 and 2 will show the updated execution  and task 3, 4, and 5 won't. But that itself could be revealing so.

Reply:
Yep that was my intent. It'll definitely be confusing if someone seeing the log line hasn't read the code carefully, but I think that's a reasonable expectation here